JMdictFiles/sameKanjiCounter.js

```diff
@@ -0,0 +1,51 @@
+const { JM } = require('./dic');
+
+const createKanjiToCountObject = (dict) => {
+    const kanjiToCount = {};
+    for (let i = 0; i < dict.length; i++) {
+        kanjiToCount[dict[i].kanji] = (kanjiToCount[dict[i].kanji] || 0) + 1;
+    }
```

Comment:
This works, although a more common practice among JS/TS developers is to make extensive use of `.forEach` , `.map` and `.reduce` functions to achieve the same result. Just make sure you can do both to understand the code that someone else might write.